src/de/mod.rs

```diff
@@ -662,6 +662,31 @@ mod tests {
         assert!(from_str::<[i32; 2]>("[0, 1,]").is_err());
     }
 
+    #[test]
+    fn tuple_variant() {
+        #[derive(Debug, Deserialize, PartialEq)]
+        enum Ops {
+            Exit(),
+            Square(i32),
+            Add(i64, i64),
+        }
+        assert_eq!(from_str(r#"{"Exit":[]}"#), Ok(Ops::Exit()));
```

Comment:
That is a bit odd format (`[]`), but I guess that is correct.

Reply:
Did you have a look at the one element case where the list suddenly becomes a number? This is odd.